translator/trace/jaeger/jaegerproto_to_protospan.go

```diff
@@ -0,0 +1,274 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jaeger
+
+import (
+	"encoding/base64"
+	"fmt"
+	"reflect"
+	"strconv"
+
+	commonpb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1"
+	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
+	model "github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/thrift-gen/jaeger"
+
+	"github.com/open-telemetry/opentelemetry-service/consumer/consumerdata"
+	"github.com/open-telemetry/opentelemetry-service/internal"
+	tracetranslator "github.com/open-telemetry/opentelemetry-service/translator/trace"
+)
+
+// ProtoBatchToOCProto converts a single Jaeger Proto batch of spans to a OC proto batch.
+func ProtoBatchToOCProto(batch model.Batch) (consumerdata.TraceData, error) {
+	ocbatch := consumerdata.TraceData{
+		Node:         jProtoProcessToOCProtoNode(batch.GetProcess()),
+		Spans:        jProtoSpansToOCProtoSpans(batch.GetSpans()),
+		SourceFormat: "jaeger", // TODO(@jpkrohling): is this appropriate here? the proto model is Jaeger's
```

Comment:
In other places, the `SourceFormat` is set inside the endpoints (`SubmitBatches()`, for instance). In this specific case, I believe it might be appropriate to have this in the translator itself, but I can change it to the receiver for consistency, if necessary.

Reply:
Please, move to the receiver for consistency.